pkg/migrate/migrate.go

```diff
@@ -1146,3 +1296,249 @@ func resetReclaimPolicy(ctx context.Context, w *log.Logger, clientset k8sclient.
 
 	return nil
 }
+
+// buildPVConsumerPod creates a pod spec for consuming a pvc
+func (p *PVMigrator) buildPVConsumerPod(pvcName string) *corev1.Pod {
+	podName := p.podNameOverride
+	if podName == "" {
+		tmp := uuid.New().String()[:5]
+		podName = fmt.Sprintf("pvmigrate-vol-consumer-%s-%s", pvcName, tmp)
+	}
+	if len(podName) > 63 {
+		podName = podName[0:31] + podName[len(podName)-32:]
+	}
+	return &corev1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: "default",
+		},
+		Spec: corev1.PodSpec{
+			RestartPolicy: corev1.RestartPolicyNever,
+			Volumes: []corev1.Volume{
+				{
+					Name: "tmp",
+					VolumeSource: corev1.VolumeSource{
+						PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+							ClaimName: pvcName,
+						},
+					},
+				},
+			},
+			Containers: []corev1.Container{
+				{
+					Name:  "busybox",
+					Image: "busybox",
+					Command: []string{
+						"sleep",
+						"3600",
+					},
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							MountPath: "/tmpmount",
+							Name:      "tmp",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// buildPVC creates a temporary PVC requesting for 1Mi of storage for a provided storage class name.
+func (p *PVMigrator) buildTmpPVC(pvc corev1.PersistentVolumeClaim, sc string) *corev1.PersistentVolumeClaim {
+	pvcName := p.pvcNameOverride
```

Comment:
instead of having pvcNameOverride, I'd generally recommend using the source PVC's name with a suffix and then trimmed

Reply:
(And name the pod based on the PVC being tested too)

Reply:
this makes things much clearer to any cluster operator that encounters this in progress

Reply:
The name override is meant for testing since we need a deterministic name.

Reply:
The name should ALWAYS be deterministic, never random

The way to do that is to take something like `my-app-pvc-0` and turn it into `my-app-pvc-0-pvcmigrate-accessmodes`

Reply:
Will make this change.